parity/run.rs

```diff
@@ -379,7 +383,7 @@ fn execute_light_impl(cmd: RunCmd, logger: Arc<RotatingLogger>) -> Result<Runnin
 
 fn execute_impl<Cr, Rr>(cmd: RunCmd, logger: Arc<RotatingLogger>, on_client_rq: Cr,
 						on_updater_rq: Rr) -> Result<RunningClient, String>
-	where Cr: Fn(String) + 'static + Send,
+	where Cr: Fn(String) -> Result<(), ()> + 'static + Send,
```

Comment:
Is this change needed? 

It breaks `parity-clib` compatibility

Reply:
OK, it'd be nice to have the RPC response reflect whether or not the restart succeeded (although not necessary) but changing the return value of the restart client callback to an integer doesn't seem very clean so I'll change back the return value of the callback for now! :+1: